tools/benchmarks/llm_eval_harness/meta_eval_reproduce/prepare_meta_eval.py

```diff
@@ -0,0 +1,176 @@
+    if model_name not in ["Meta-Llama-3.1-8B-Instruct","Meta-Llama-3.1-70B-Instruct","Meta-Llama-3.1-405B-Instruct"]:
+        raise ValueError("Only Meta-Llama-3.1-8B-Instruct, Meta-Llama-3.1-70B-Instruct, Meta-Llama-3.1-405B-Instruct models are supported for IFEval")
+    original_dataset_name = "wis-k/instruction-following-eval"
+    meta_dataset_name = f"meta-llama/{model_name}-evals"
+    meta_data = load_dataset(
+        meta_dataset_name,
+        name=f"{model_name}-evals__ifeval__strict__details",
+        split="latest"
+        )
+    ifeval_data = load_dataset(
+        original_dataset_name,
+        split="train"
+        )
+    meta_data = meta_data.map(get_question)
+    meta_df = meta_data.to_pandas()
+    ifeval_df = ifeval_data.to_pandas()
+    ifeval_df = ifeval_df.rename(columns={"prompt": "input_question"})
+
+    joined = meta_df.join(ifeval_df.set_index('input_question'),on="input_question")
+    joined = joined.rename(columns={"input_final_prompts": "prompt"})
+    joined = joined.rename(columns={"is_correct": "previous_is_correct"})
+    joined = Dataset.from_pandas(joined)
+    joined = joined.select_columns(["input_question", "prompt", "previous_is_correct","instruction_id_list","kwargs","output_prediction_text","key"])
+    joined.rename_column("output_prediction_text","previous_output_prediction_text")
+    for item in joined:
+        check_sample(item)
+    joined.to_parquet(output_dir + f"/joined_ifeval.parquet")
+def get_math_data(model_name,output_dir):
+    print(f"preparing the math data using {model_name}'s evals dataset")
+    if model_name not in ["Meta-Llama-3.1-8B-Instruct","Meta-Llama-3.1-70B-Instruct","Meta-Llama-3.1-405B-Instruct"]:
+        raise ValueError("Only Meta-Llama-3.1-8B-Instruct, Meta-Llama-3.1-70B-Instruct, Meta-Llama-3.1-405B-Instruct models are supported for MATH_hard")
+    original_dataset_name = "lighteval/MATH-Hard"
+    meta_dataset_name = f"meta-llama/{model_name}-evals"
+    meta_data = load_dataset(
+        meta_dataset_name,
+        name=f"{model_name}-evals__math_hard__details",
+        split="latest"
+        )
+    math_data = load_dataset(
+        original_dataset_name,
+        split="test"
+        )
+    meta_df = meta_data.to_pandas()
+    math_df = math_data.to_pandas()
+    math_df = math_df.rename(columns={"problem": "input_question"})
+
+    joined = meta_df.join(math_df.set_index('input_question'),on="input_question")
+    joined = Dataset.from_pandas(joined)
+    joined = joined.select_columns(["input_question", "input_correct_responses", "input_final_prompts", "is_correct","solution","output_prediction_text"])
+    joined = joined.rename_column("is_correct","previous_is_correct")
+    joined = joined.rename_column("output_prediction_text","previous_output_prediction_text")
+    for item in joined:
+        check_sample(item)
+    joined.to_parquet(output_dir + f"/joined_math.parquet")
+def get_question(example):
```

Comment:
some code formating would make the code easier to read / follow

Reply:
Thanks, code has been formatted